docs/plugins/building_a_plugin/_layer_data_guide.md

```diff
@@ -56,6 +56,11 @@ class ArrayLike(Protocol):
 # typing than literal array types (e.g. numpy, dask, xarray, etc...)
 ```
 
+Note that when writing your own plugin, such type annotations are nearly always optional, 
+except in rare cases like `magicgui` function widgets.
+Here, for clarity, we've provided type names which indicate how values of the type are 
+being used and the level of generality that's possible (e.g., with `ArrayLike`). 
```

Comment:
Thanks. What do you mean "how values of the type are being used"? As in the definition of each data type and their relationship with each other?

Reply:
I mean to refer to the meaning of the values of these types in the context of a napari plugin, rather than just generally. Like I think it's quite helpful that `Dict` is renamed as `LayerProps` to show the role that data in the dict are playing when used in the plugin, and that `FullDataLayer` relabels an otherwise relatively complex tuple type with a name that tells readers / plugin developers clearly how those data fit in the functioning of the plugin.

Maybe something like "the role of the corresponding values in a plugin" replacing "how values of the type are being used" would make this snippet more readable?

Reply:
Ohhh I see what you are saying. You're right. I like your suggested wording!

It was not immediately clear to me that `LayerProps` stood for 'layer properties' actually.